src/core/dbus-execute.c

```
@@ -933,7 +1004,10 @@ int bus_set_transient_exec_command(
 
                         c->argv = TAKE_PTR(argv);
 
-                        c->flags = b ? EXEC_COMMAND_IGNORE_FAILURE : 0;
+                        if (!is_ex_prop)
+                                c->flags = b ? EXEC_COMMAND_IGNORE_FAILURE : 0;
+                        else
+                                c->flags = exec_command_flags_from_strv(ex_opts);
```

if the user specifies an invalid flag we should return a failure here